DevoxxClientMobile/src/main/java/com/devoxx/views/SponsorPresenter.java

```diff
@@ -26,100 +26,161 @@
 
     @FXML
-    private Region imageSpacer;
-
-    @FXML
-    private Label name;
-
+    private View sponsorView;
+    
     @FXML
-    private Label details;
+    private VBox content;
 
     @FXML
-    private ImageView imageView;
+    private PasswordField password;
 
     @FXML
-    private ImageView logo;
-
-    private FloatingActionButton webLaunchFAB;
+    private Label message;
     
-    // sponsorProperty
-    private final ObjectProperty<Sponsor> sponsorProperty = new SimpleObjectProperty<Sponsor>(this, "sponsor") {
-        @Override protected void invalidated() {
-            Sponsor sponsor = get();
-            logo.setImage(ImageCache.get(sponsor.getPicture()));
-            name.setText(sponsor.getName());
-            
-            String url = sponsor.getSummary();
-            webLaunchFAB.setVisible(url != null && !url.isEmpty());
-
-            details.setText(sponsor.getSummary());
-            resizeImages();
-        }
-    };
-    public final ObjectProperty<Sponsor> sponsorProperty() {
-        return sponsorProperty;
-    }
-    public final Sponsor getSponsor() {
-        return sponsorProperty.get();
-    }
-    public final void setSponsor(Sponsor value) {
-        sponsorProperty.set(value);
-    }
+    @Inject
+    private Service service;
 
+    private CharmListView<SponsorBadge, String> lvBadges;
+    private String name;
+    private String slug;
 
     public void initialize() {
-        webLaunchFAB = Util.createWebLaunchFAB(() -> getSponsor().getSummary());
-        sponsor.getLayers().add(webLaunchFAB.getLayer());
 
-        sponsor.setOnShowing( event -> {
+        sponsorView.setOnShowing(event -> {
             AppBar appBar = getApp().getAppBar();
-            appBar.setNavIcon(getApp().getNavBackButton());
+            appBar.setNavIcon(getApp().getNavMenuButton());
             appBar.setTitleText(DevoxxView.SPONSOR.getTitle());
+            checkAndLoadView();
+        });
+        
+        sponsorView.setOnHiding(event -> {
+            sponsorView.getLayers().clear();
+        });
+    }
 
-            // give images full width of scene
-            sponsor.getScene().widthProperty().addListener((observable, oldValue, newValue) -> resizeImages());
-            sponsor.getScene().heightProperty().addListener((observable, oldValue, newValue) -> resizeImages());
+    public void setSponsor(String name, String slug) {
+        this.name = name;
+        this.slug = slug;
+        checkAndLoadView();
+    }
 
-            // randomly change image on each showing
-            imageView.setImage(Util.getMediaBackgroundImage());
+    @FXML
+    private void signIn() {
+        final GluonObservableObject<String> passwordObject = service.authenticateSponsor(password.getText());
+        passwordObject.stateProperty().addListener((o, ov, nv) -> {
+            if (nv == ConnectState.SUCCEEDED) {
```

Comment:
When the remote function call succeeded, the provided password should be validated here with the returned value.

Reply:
Ouch! Thanks for pointing it out.